bin/release

```diff
@@ -0,0 +1,126 @@
+"
+}
+
+# Bump a site's release according to its changelog.
+#
+# Usage: release SITE IS_MAJOR SHOULD_COMMIT
+#
+#  SITE           string  name of a folder in the "sites" directory
+#  IS_MAJOR       0|1     whether we should force a major release (breaking change)
+#  SHOULD_COMMIT  0|1     whether we should commit the release to the git log
+function release() {
+    site=$1
+    is_major=$2
+    should_commit=$3
+
+    changelog=$(sed "/## \[Unreleased/,/## \[/!d" "${SITES_DIRECTORY}/${site}/CHANGELOG.md" | tail -n +2)
+    # Extract the last version released from changelog
+    current_version=$(echo "$changelog" | grep "## \[" | sed -E "s/.*\[(.*)\].*/\1/")
+    # Split current version to array with major/minor/revision release numbers
+    IFS=\. read -r -a current_version_array <<<"${current_version}"
+
+    # Bump version to above major, minor or revision number depending on changelog content
+    if [[ "${is_major}" == 1  ]]; then
+        # Major release
+        major_release=$((current_version_array[0] + 1))
+        new_version="${major_release}.0.0"
+    else
+        if ! echo "${changelog}" | grep "###" &> /dev/null; then
+            # No changes since the last release, do nothing
+            echo -e "${COLOR_INFO}Nothing to release on ${site}${COLOR_RESET}"
+            return 0
+        fi
+        if echo "${changelog}" | grep "###" | grep -v "Fixed" &> /dev/null; then
+            # Minor release
+            minor_release=$((current_version_array[1] + 1))
+            new_version="${current_version_array[0]}.${minor_release}.0"
+        else
+            # Revision release (fix)
+            revision_release=$((current_version_array[2] + 1))
+            new_version="${current_version_array[0]}.${current_version_array[1]}.${revision_release}"
+        fi
+    fi
+
+    echo -e "${COLOR_INFO}Creating release v${new_version} for ${site}${COLOR_RESET}"
+
+    # Update CHANGELOG.md
+    # - Add tag and date for the new releave in the form: ## [2.10.2]  2020-10-07
+    sed -i -E "/Unreleased/a \\\n## [${new_version}] - $(date +'%Y-%m-%d')" "${SITES_DIRECTORY}/${site}/CHANGELOG.md"
+    # - Add release link in footer
+    sed -i "s/^\[unreleased\].*$/[unreleased]: https:\/\/github.com\/openfun\/richie-site-factory\/compare\/${site}-${new_version}...HEAD/g" "${SITES_DIRECTORY}/${site}/CHANGELOG.md"
+    sed -i -E "/^\[${current_version}\]/i [${new_version}]: https://github.com/openfun/richie-site-factory/compare/${site}-${current_version}...${site}-${new_version}" "${SITES_DIRECTORY}/${site}/CHANGELOG.md"
+
+    # Update richie backend and frontend versions
+    sed -i -E "s/(__version__ = \")(.*)\"/\1${new_version}\"/" "${SITES_DIRECTORY}/${site}/src/backend/${site}/__init__.py"
+    sed -i -E "s/(\"version\": \")(.*)\"/\1${new_version}\"/" "${SITES_DIRECTORY}/${site}/src/frontend/package.json"
+
+    if [[ "${should_commit}" == 1 ]]; then
+        # Commit changes
+        git add "${SITES_DIRECTORY}/${site}"
```

Comment:
```suggestion
        git add "${SITES_DIRECTORY}/${site}/CHANGELOG.md"
        git add "${SITES_DIRECTORY}/${site}/src/backend/${site}/__init__.py"
        git add "${SITES_DIRECTORY}/${site}/src/frontend/package.json"
```

Instead of adding the entire site directory, I suggest to add only the files that we have updated.

Reply:
Very good idea :medal_sports: